apps/src/lib/node/ledger/ethereum_node/events.rs

```diff
@@ -711,6 +687,43 @@ pub mod eth_events {
             Ok(addrs)
         }
 
+        fn parse_transfer_to_namada_array(
+            self,
+        ) -> Result<Vec<TransferToNamada>> {
+            let array = if let Token::Array(array) = self {
+                array
+            } else {
+                return Err(Error::Decode(format!(
+                    "Expected type `Array`, got {:?}",
+                    self
+                )));
+            };
+            let mut transfers = vec![];
+            for token in array.into_iter() {
+                let transfer = token.parse_transfer_to_namada()?;
+                transfers.push(transfer);
+            }
+            Ok(transfers)
+        }
+
+        fn parse_transfer_to_namada(self) -> Result<TransferToNamada> {
+            if let Token::Tuple(tuple) = self {
+                let asset = tuple[0].clone().parse_eth_address()?;
```

Comment:
I'm surprised we need a `clone` here. The function has owner of tuple and it's not used again elsewhere

Reply:
Calling `parse_` directly on `tuple`'s elements won't work as `tuple` still has ownership of the individual elements (though the `parse_` methods want ownership as well). Have updated the code to rename `tuple` to `items` (since it's actually a `Vec`), and avoid `clone` by moving items out of the vec first before calling the `.parse_` methods, so that they can take ownership.